Python 3 compatibility

fnkr opened the pull request.

fnkr committed:
Python 3 compatibility

stephen-soltesz commented:
@fnkr -- thank you for the patch!

To make sure travis also runs tests using python3, would you be willing to add the build matrix for python to .travis.yml?

It should look like: and pass in travis.
```
language: python
python:
- '2.7'
- '3.5'
```

With that change, lgtm.

fnkr commented:
@stephen-soltesz I've updated the Travis config and fixed all failing tests but one. I couldn't figure out how to fix the last failing test, maybe you can take a look at it? You have push access to my branch.

stephen-soltesz committed:
Sort dict keys for deterministic label order

stephen-soltesz commented:
@fnkr -- thank you again for the additional changes. That turned out to be more than I expected to enable python3 unit testing.

The last issue was due to the random order `dict.items()` returns keys. Making sure they're always `sorted` first appears to resolve it.
